yb-voyager/src/migassessment/sizing.go

```diff
@@ -863,6 +875,151 @@ func calculateTimeTakenAndParallelJobsForImport(tableName string, dbObjects []So
+Parameters:
+
+	experimentDB: Connection to the experiment database
+	vCPUPerInstance: Number of virtual CPUs per instance.
+	memPerCore: Memory per core.
+
+Returns:
+
+	[]ExpDataShardedLoadTime: A slice containing the fetched sharded load time information.
+	error: Error if any.
+*/
+func getExpDataShardedLoadTime(experimentDB *sql.DB, vCPUPerInstance int, memPerCore int) ([]ExpDataShardedLoadTime, error) {
+	selectQuery := fmt.Sprintf(`
+		SELECT csv_size_gb, 
+			   migration_time_secs, 
+			   parallel_threads 
+		FROM %v 
+		WHERE num_cores = ? 
+			AND mem_per_core = ?
+		ORDER BY scsv_size_gb;
+	`, SHARDED_LOAD_TIME_TABLE)
+	rows, err := experimentDB.Query(selectQuery, vCPUPerInstance, memPerCore)
+
+	if err != nil {
+		return nil, fmt.Errorf("error while fetching load time info with query [%s]: %w", selectQuery, err)
+	}
+	defer func() {
+		if closeErr := rows.Close(); closeErr != nil {
+			log.Warnf("failed to close result set for query: [%s]", selectQuery)
+		}
+	}()
+
+	var shardedLoadTimes []ExpDataShardedLoadTime
+	for rows.Next() {
+		var shardedLoadTime ExpDataShardedLoadTime
+		if err = rows.Scan(&shardedLoadTime.csvSizeGB, &shardedLoadTime.migrationTimeSecs,
+			&shardedLoadTime.parallelThreads); err != nil {
+			return nil, fmt.Errorf("cannot fetch data from experiment data table with query [%s]: %w", selectQuery, err)
+		}
+		shardedLoadTimes = append(shardedLoadTimes, shardedLoadTime)
+	}
+	return shardedLoadTimes, nil
+}
+
+/*
+findClosestRecordFromExpDataShardedLoadTime finds the closest record from the experiment data for the size of the table.
+Parameters:
+
+	loadTimes: A slice containing the fetched sharded load time information.
+	objectSize: The size of the table in gigabytes.
+
+Returns:
+
+	ExpDataShardedLoadTime: The closest record from the experiment data for the size of the table which is closest to the object size.
+*/
+func findClosestRecordFromExpDataShardedLoadTime(loadTimes []ExpDataShardedLoadTime, objectSize float64) ExpDataShardedLoadTime {
+	closest := loadTimes[0]
+	minDiff := math.Abs(objectSize - closest.csvSizeGB.Float64)
+
+	for _, num := range loadTimes {
+		diff := math.Abs(objectSize - num.csvSizeGB.Float64)
+		if diff < minDiff {
+			minDiff = diff
+			closest = num
+		}
+	}
+
+	return closest
+}
+
+/*
+getMultiplicationFactorForImportTimeBasedOnIndexes calculates the multiplication factor for import time based on number
+of indexes on the table.
+
+Parameters:
+
+	table: Metadata for the database table for which the multiplication factor is to be calculated.
+	sourceIndexMetadata: A slice containing metadata for the indexes in the database.
+
+Returns:
+
+	float64: The multiplication factor for import time based on the number of indexes on the table.
+*/
+func getMultiplicationFactorForImportTimeBasedOnIndexes(table SourceDBMetadata, sourceIndexMetadata []SourceDBMetadata) float64 {
+	var numberOfIndexesOnTable float64 = 0
```

Comment:
We should not hardcode this. We should have the index numbers in database and use them for calculation. Things can change release over release

Reply:
We cam merge this PR for Dev build. But the change has to happen before the release

Reply:
sure. Created another task for this.